src/kubernetes.cc

```diff
@@ -1042,27 +1043,47 @@ json::value KubernetesReader::FindTopLevelOwner(
 #endif
   const json::Object* metadata = obj->Get<json::Object>("metadata");
 #ifdef VERBOSE
-  LOG(DEBUG) << "FindTopLevelOwner: metadata is " << *metadata;
+  LOG(DEBUG) << "FindTopLevelController: metadata is " << *metadata;
 #endif
   if (!metadata->Has("ownerReferences")) {
 #ifdef VERBOSE
-    LOG(DEBUG) << "FindTopLevelOwner: no owner references in " << *metadata;
+    LOG(DEBUG) << "FindTopLevelController: no owner references in "
+               << *metadata;
 #endif
     return object;
   }
   const json::Array* refs = metadata->Get<json::Array>("ownerReferences");
 #ifdef VERBOSE
-  LOG(DEBUG) << "FindTopLevelOwner: refs is " << *refs;
+  LOG(DEBUG) << "FindTopLevelController: refs is " << *refs;
 #endif
-  if (refs->size() > 1) {
-    LOG(WARNING) << "Found multiple owner references for " << *obj
+  std::vector<const json::Object*> controller_refs;
+  for (const json::value& ref : *refs) {
+    const json::Object* ref_obj = ref->As<json::Object>();
+    if (ref_obj->Has("controller") &&
+        ref_obj->Get<json::Boolean>("controller")) {
+      controller_refs.emplace_back(ref_obj);
+    }
+  }
+#ifdef VERBOSE
+  LOG(DEBUG) << "FindTopLevelController: contoller_refs is "
+             << controller_refs;
+#endif
+  if (controller_refs.size() == 0) {
+#ifdef VERBOSE
+    LOG(DEBUG) << "FindTopLevelController: no controller owner references in "
+               << *metadata;
+#endif
+    return object;
+  }
+  if (controller_refs.size() > 1) {
```

Comment:
Being that we're specifically looking for controllers now, this is from the Kubernetes documentation https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#objectmeta-v1-meta.

"There cannot be more than one managing controller."

How do you feel about removing this check?

Reply:
Correct - this was more for defensive programming, just in case.

If we want to ignore any controller after the first one, I can simplify the for loop above to store just the first controller, and break after it finds one. WDYT?

Reply:
Yes, let's just declare a single `const json::Object* controller_ref = nullptr` with a comment that Kubernetes objects are supposed to have at most one and a link to the doc that Bryan referenced. Then break out of the loop when you find one.

Reply:
Done